daemons/fenced/pacemaker-fenced.c

```diff
@@ -772,6 +794,10 @@ cib_devices_update(void)
         }
     }
 
+    /* have list repopulated if cib has a watchdog-fencing-resource */
+    g_list_free_full(stonith_watchdog_targets, free);
+    stonith_watchdog_targets = NULL;
```

Comment:
Not necessarily for this PR, but I think to address the timing issue you described, we should keep a temporary targets list while it's being built up, then replace the main one with the final result. Maybe add a todo comment

Reply:
added a TODO